arbor/util/rangeutil.hpp

```diff
@@ -297,11 +302,14 @@ bool is_sorted(const Seq& seq) {
 template <
     typename Seq,
     typename Proj,
-    typename Compare = std::less<typename std::result_of<Proj (typename sequence_traits<const Seq&>::value_type)>::type>
+    typename Compare = std::less<std::result_of_t<Proj (typename sequence_traits<const Seq&>::value_type)>>
 >
 bool is_sorted_by(const Seq& seq, const Proj& proj, Compare cmp = Compare{}) {
-    auto i = std::begin(seq);
-    auto e = std::end(seq);
+    using std::begin;
+    using std::end;
```

Comment:
Same here? Is this some sort of ADL shenanigans?

Reply:
Yep, it's exactly ADL shenanigans. `begin` is a customization point, and the 'right way' to get the beginning iterator is through this sort of construction.